projects/ccf-shared/src/lib/components/tracking-popup/tracking-popup.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, ElementRef, Inject } from '@angular/core';
+import { MAT_SNACK_BAR_DATA } from '@angular/material/snack-bar';
+import { TrackingState } from '../../services/globals/tracking.state'; 
+
+@Component({
+  selector: 'ccf-tracking-popup',
+  templateUrl: './tracking-popup.component.html',
+  styleUrls: ['./tracking-popup.component.scss']
+})
+export class TrackingPopupComponent {
+
+  container: HTMLElement;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  constructor(elementRef: ElementRef<HTMLElement>, readonly tracking: TrackingState, @Inject(MAT_SNACK_BAR_DATA) public data: any) {
+    this.container = elementRef.nativeElement;
+  }
+
+  dismiss(): void {
+    this.data.preClose();
+  }
+
+  submit(entry: boolean): void {
+    this.tracking.setAllowTelemetry(entry);
+    this.dismiss();
+  }
+
+  showOptIn(): boolean {
```

Comment:
The logic of showOptIn / showOptOut can be greatly simplified. I could tell you the one liner for each function, but please do look closely at the logic and see for yourself.